app/src/main/java/com/kamron/pogoiv/scanlogic/IVScanResult.java

```diff
@@ -335,4 +336,17 @@ public void refineByEggRaid() {
         }
         iVCombinations = refinedList;
     }
+
+    public void addPossibilitiesFrom(IVScanResult ivs) {
+
+        for (IVCombination ivc : ivs.iVCombinations) {
```

Comment:
To automatically avoid duplicated IV combinations, it is easier to change the type of iVCombinations from
`public ArrayList<IVCombination> iVCombinations = new ArrayList<>();`
To:
`public HashSet<IVCombination> iVCombinations = new HashSet<>();`

The latter requires that IVCombination class has a proper implementation of hashCode().
Looking at the code it seems it already has one, so it should work out of the box.

Reply:
We need to be able to sort the IVCombinations to display them correctly in the "show all" screen. As far as I know, the set does not support sorting?

Reply:
You're right, the sorting can be done only when displaying all results, with a simple ArrayList allocation and sorting, i.e.:
```java
List sortedList = new ArrayList(yourHashSet);
Collections.sort(sortedList);
```
This is not high priority but HashSet should be faster than ArrayList, and sorting can be delayed and done only if the user shows all combinations. Can be an idea for future development!

Reply:
You're right, that'd probably be a better solution. We'd need to maybe change a little in IVResultsAdapter so it doesnt take an IVScanresult and instead take a collection of IVs instead though.